### ISSUE TITLE
Can Webp-Hero convert by MIME Type?

### ISSUE THREAD

**Issue by joshuawaheed** [chase-moskal/webp-hero#36]
When implementing this in my project, I discovered that this plugin updates WebP images if their file extension is ".webp", which is great.

The problem I have now is that I have an image coming from an external API, which has a MIME Type of "image/webp" but its file extension is ".png", which isn't being converted and is still showing a broken image in IE11 and Safari.

Is there a method that looks at the MIME Type instead of the file extension? Or is this something that could potentially be implemented please?

**Comment by chase-moskal** [comment]
yes, i'd love this to be implemented

we're currently blocked by being unable to build google's libwebp library, see https://github.com/chase-moskal/webp-hero/issues/31#issuecomment-630490492

hopefully at some point i can take another stab at this, get a fresh build, and implement some backlogged features

once we get the libwebp build working, i'd like to do a serious refactor of webp-hero and include features like the topic of this issue :+1:

**Comment by chase-moskal** [comment]
closing this issue as stale

there is a new note in webp-hero's readme about this deficiency

anybody: please open a new issue or pull request if you're interested to collaborate on this feature
